ability_item_usage_zuus.lua

```diff
@@ -451,90 +457,132 @@ Consider[3] = function()
 	return BOT_ACTION_DESIRE_NONE
 end
 
-local function DontSteal(enemy)
-	return AbilityExtensions:GetUnitList(UNIT_LIST_ALLIED_HEROES):Remove(npcBot):Any(function(t) return enemy:
-			WasRecentlyDamagedByHero(t, 1.5)
-	end) and not enemy:WasRecentlyDamagedByHero(npcBot, 2.5)
-end
+function DebugDrawUltTable()
+	if debugmode then
 
-local function ShouldUseUltimate(enemy)
-	return CanCast[6](enemy) and DontSteal(enemy)
-end
+		local drawY = 720
+		local drawStep = 18
 
-Consider[6] = function()
+		for enemyName, enemyStats in pairs(npcBot.ult) do
+			local timeElapsed = DotaTime() - enemyStats[1]
+			local enemyBot = enemyStats[2]
+			local enemyHero = enemyStats[3]
+			local wasKillable = enemyStats[4]
+			local isNull = ""
 
-	local ability = AbilitiesReal[6]
-	if not ability:IsFullyCastable()
-	then
-		return BOT_ACTION_DESIRE_NONE
-	end
+			if enemyBot:IsNull() then

```

Comment:
Is there a function IsNull() in Lua language? The general usage seems to be `enemyBot == nil`

Reply:
I think IsNull() comes from the C interface.  Wherever it's from, it detects when a bot object _exists_ but is no longer valid.  From what I can see, when a bot goes out of vision, its object loses all of its API functions, _but is not nil_.  E.g. calling enemyBot:GetUnitName will return an error. This is only really an issue when keeping track of bots across frames.

I assume the reason it works this way is to prevent you from keeping track of the real hero by just tracking the object id.  Interestingly, it seems dying does not cause the object to be cleared.

Reply:
I get it, this is really interesting.